lib/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java

```diff
@@ -240,13 +242,28 @@ public String getCriPrivateApiKey(String criId) {
 
     public CredentialIssuerConfig getCredentialIssuerActiveConnectionConfig(
             String credentialIssuerId) {
-        String activeConnection = getActiveConnection(credentialIssuerId);
+        return getCriConfigForConnection(
+                getActiveConnection(credentialIssuerId), credentialIssuerId);
+    }
+
+    public CredentialIssuerConfig getCriConfig(CriOAuthSessionItem criOAuthSessionItem) {
+        return getCriConfigForConnection(
+                criOAuthSessionItem.getConnection(), criOAuthSessionItem.getCriId());
+    }
+
+    public CredentialIssuerConfig getCriConfigForConnection(String connection, String criId) {
         final String pathTemplate =
                 ConfigurationVariable.CREDENTIAL_ISSUERS.getPath() + "/%s/connections/%s";
-        Map<String, String> result =
-                getSsmParameters(pathTemplate, false, credentialIssuerId, activeConnection);
+        Map<String, String> result = getSsmParameters(pathTemplate, false, criId, connection);
+
+        if (result == null || result.isEmpty()) {
+            throw new NoConfigForConnectionException(
```

Comment:
Wouldn't it be better if we catch this exception in lambda and return it later with a specific error code and this message?